Purchases/Purchasing/StoreKitAbstractions/PromotionalOffer.swift

```diff
@@ -34,15 +34,25 @@ extension PromotionalOffer {
                                  timestamp: self.timestamp as NSNumber)
     }
 
+    /// - Throws: ``ErrorCode/unexpectedBackendResponseError`` if the signature cannot be encoded in UTF8
     @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
     var sk2PurchaseOption: Product.PurchaseOption {
-        return .promotionalOffer(
-            offerID: self.identifier,
-            keyID: self.keyIdentifier,
-            nonce: self.nonce,
-            signature: self.signature.data(using: .utf8)!,
-            timestamp: self.timestamp
-        )
+        get throws {
+            guard let signatureData = self.signature.data(using: .utf8) else {
+                throw ErrorUtils.unexpectedBackendResponse(
+                    withSubError: nil,
```

Comment:
What's your thought on creating a suberror, like `PromotionalOfferDataError` or something? I kinda think anytime we want to add extra context, we should consider creating a specific suberror.

Reply:
Yeah I can't wait to basically have an `Error` type for every layer (`ErrorCode` at the top, `PurchasesOrchestrator`, `HTTPClient`, etc.), and make sure that every layer only throws error types of that layer. Right now we might be throwing error types from several layers down which can be hard to track down.

I'll change this!

Reply:
error types for every layer would be an absolutely fantastic cleanup